proton-j/src/main/java/org/apache/qpid/proton/message/Message.java

```diff
@@ -210,4 +213,55 @@ public static Message create(Header header,
     void clear();
 
     MessageError getError();
+
+    /**
+     * @return the total number of body {@link Section} elements contained in this {@link Message}
+     */
+    int getBodySectionCount();
+
+    /**
+     * Sets the body {@link Section} instances to use when encoding this message.  The value
```

Comment:
Perhaps note that the changes to the collection afterwards aren't reflected in the message?